src/packagedef.jl

```diff
@@ -910,13 +914,13 @@ function track(mod::Module, file::AbstractString; mode=:sigs, kwargs...)
             CodeTracking._pkgfiles[id] = pkgdata.info
         end
         push!(pkgdata, relpath(file, pkgdata)=>FileInfo(fm))
-        init_watching(pkgdata, (file,))
+        init_watching(pkgdata, (String(file),))
```

Comment:
If `file` is only known to be `Any`, this should be `String(file)::String`. Sadly there is no guarantee that constructors return an object of the given type, https://github.com/JuliaLang/julia/issues/42372.

Reply:
You marked this as resolved but didn't implement the suggested change. Please don't do that.